src/cli/study-backrefs.js

```diff
@@ -0,0 +1,358 @@
+  "resource-timing-1": "resource-timing",
+  "wai-aria-1.1": "wai-aria-1.2"
+};
+
+// TODO: check the link is non-normative (somehow)
+const shortnameOfNonNormativeDocs = [
+  "aria-practices", "rdf11-primer", "discovery-api", "wake-lock-use-cases", "capability-urls", "streamproc", "media-source-testcoverage", "using-aria", "wai-aria-practices-1.1", "sensor-polyfills", "sensors", "design-principles", "security-questionnaire", "css3-preslev", "clreq", "klreq", "typography", "ssml-sayas", "css3-marquee", "spatial-navigation", "ilreq", "css-print", "books", "dpub-pagination", "predefined-counter-styles", "css-2017", "sniffly", "wai-aria-practices-1.2", "wai-aria-implementation", "wai-aria-roadmap", "wai-aria-practices", "webdatabase", "installable-webapps", "motion-sensors", "file-system-api", "media-accessibility-reqs", "webrtc-interop-reports", "webaudio-usecases", "web-audio-perf", "Audio-EQ-Cookbook", "web-intents", "touch-events-extensions", "fingerprinting-guidance", "webrtc-nv-use-cases", "html-design-principles", "storage-partitioning", "jlreq", "accept-encoding-range-test", "security-privacy-questionnaire", "dpub-latinreq"
+];
+
+const report = {};
+
+function recordAnomaly(spec, anomalyType, link) {
+  if (!report[spec.url]) {
+    report[spec.url] = {
+      title: spec.title,
+      notExported: [],
+      notDfn: [],
+      brokenLink: [],
+      evolvingLink: [],
+      outdatedSpec: [],
+      unknownSpec: [],
+      datedUrl: []
+    };
+  }
+  report[spec.url][anomalyType].push(link);
+}
+
+const missingSpecs = {};
+
+function recordUnknownSpec(link, shortname, spec) {
+  if (!missingSpecs[shortname]) {
+    missingSpecs[shortname] = [];
+  }
+  missingSpecs[shortname].push({link, spec: spec.url});
+}
+
+function studyCrawlResults(edResults, trResults) {
+  edResults.forEach(spec => {
+    Object.keys(spec.links)
+      .filter(matchSpecUrl)
+      .forEach(l => {
+        let shortname;
+        let nakedLink = l.replace(/#.*$/, '');
+        if (nakedLink.endsWith(".html")) {
+          nakedLink = nakedLink.replace(/\/[^/]*\.html/, '/');
+        }
+        if (nakedLink[nakedLink.length - 1] !== '/') {
+          nakedLink += '/';
+        }
+        shortname = (edResults.find(r => r.url === nakedLink || (r.release && r.release.url === nakedLink) || r.nightly.url === nakedLink || (r.series && nakedLink === 'https://www.w3.org/TR/' + r.series.shortname + '/') ) || {}).shortname;
+        if (!shortname) {
+          try {
+            ({shortname} = computeShortname(l));
+          } catch (e) {
+            let m = l.match(/www\.w3\.org\/TR\/[0-9]{4}\/[A-Z]+-(.+)-[0-9]{8}/);
+            if (m) {
+              // Ignore links to previous versions
+              if (m[1] !== spec.shortname) {
+                recordAnomaly(spec, "datedUrl", l);
+              }
+            } else {
+              recordAnomaly(spec, "unknownSpec", l);
+            }
+          }
+        }
+        if (shortnameMap[shortname]) {
```

Comment:
These checks should only be meaningful when `computeShortname` is called. I suggest to move them there for clarity.

By definition, the contents of the crawl report should only contain "good" shortnames since we're going to use the crawl list as source to look for the spec.